internals/overlord/logstate/opentelemetry/opentelemetry_test.go

```diff
@@ -0,0 +1,385 @@
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [
+				{
+					"timeUnixNano": 1704026090000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #1"}
+				},
+				{
+					"timeUnixNano": 1704026092000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #3"}
+				},
+				{
+					"timeUnixNano": 1704026094000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #5"}
+				},
+				{
+					"timeUnixNano": 1704026097000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #8"}
+				}
+			]
+		}]
+	},{
+		"resource": {
+			"attributes": [{"key": "service.name", "value": {"stringValue": "svc2"}}]
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [
+				{
+					"timeUnixNano": 1704026091000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #2"}
+				},
+				{
+					"timeUnixNano": 1704026096000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #7"}
+				}
+			]
+		}]
+	},{
+		"resource": {
+			"attributes": [{"key": "service.name", "value": {"stringValue": "svc3"}}]
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [
+				{
+					"timeUnixNano": 1704026093000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #4"}
+				},
+				{
+					"timeUnixNano": 1704026095000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #6"}
+				}
+			]
+		}]
+	},{
+		"resource": {
+			"attributes": [{"key": "service.name", "value": {"stringValue": "svc4"}}]
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [{
+				"timeUnixNano": 1704026098000000000,
+				"severityText": "",
+				"severityNumber": 0,
+				"body": {"stringValue": "log line #9"}
+			}]
+		}]
+	}
+]}`)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		c.Assert(r.Method, Equals, http.MethodPost)
+		c.Assert(r.Header.Get("Content-Type"), Equals, "application/json")
+
+		reqBody, err := io.ReadAll(r.Body)
+		c.Assert(err, IsNil)
+		c.Assert(string(reqBody), DeepEquals, string(expected))
+	}))
+	defer server.Close()
+
+	client := opentelemetry.NewClient(&plan.LogTarget{Location: server.URL})
+	client.SetLabels("svc1", map[string]string{})
+	client.SetLabels("svc2", map[string]string{})
+	client.SetLabels("svc3", map[string]string{})
+	client.SetLabels("svc4", map[string]string{})
+	for _, entry := range input {
+		err := client.Add(entry)
+		c.Assert(err, IsNil)
+	}
+
+	err := client.Flush(context.Background())
+	c.Assert(err, IsNil)
```

Comment:
If the request goes through, we're asserting in the handler that it's the expected string, but we're not actually asserting that 1 request did go through. We should have a numRequests variable that's incremented in the handler and asserted on here.